pkg/microservice/aslan/core/common/repository/models/workflow_v3.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2021 The KodeRover Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package models
+
+import (
+	"github.com/koderover/zadig/pkg/types"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type WorkflowV3 struct {
+	ID          primitive.ObjectID       `bson:"_id,omitempty"  json:"id,omitempty"`
+	Name        string                   `bson:"name"           json:"name"`
+	ProjectName string                   `bson:"project_name"   json:"project_name"`
+	Description string                   `bson:"description"    json:"description"`
+	Parameters  []*ParameterSetting      `bson:"parameters"     json:"parameters"`
+	SubTasks    []map[string]interface{} `bson:"sub_tasks"      json:"sub_tasks"`
+	CreatedBy   string                   `bson:"created_by"     json:"created_by"`
+	CreateTime  int64                    `bson:"create_time"    json:"create_time"`
+	UpdatedBy   string                   `bson:"updated_by"     json:"updated_by"`
+	UpdateTime  int64                    `bson:"update_time"    json:"update_time"`
+}
+
+type ParameterSetting struct {
+	// External type parameter will NOT use this key.
+	Key string `bson:"key" json:"key"`
+	// Type list：
+	// string
+	// choice
+	// external
+	Type string `bson:"type" json:"type"`
```

Comment:
If the value is an enumeration type, we can consider this form [link](https://www.sohamkamani.com/golang/enums/#adding-string-methods-to-enum-types)